src/Sarif/NotYetAutoGenerated/SarifRewritingVisitor.cs

```diff
@@ -594,7 +594,14 @@ public virtual Resources VisitResources(Resources node)
                         var value = node.Rules[key];
                         if (value != null)
                         {
-                            node.Rules[key] = VisitNullChecked(value);
+                            ISarifNode sarifNode = value as ISarifNode;
```

Comment:
>ISarifNode [](start = 28, length = 10)

This one-off code required as IRule itself doesn't require ISarifNode to implement. All this means 1) we need special-casing to emit this method. 2) a class that implements IRule needs to also implement ISarifNode in order for the visitor to walk it successfully. Neither is very desirable.